### PR TITLE
Add Scotto36 Keyboard

### PR DESCRIPTION
**Pull request by joe-scotto**
Add Scotto36 Keyboard.

Adds a new keyboard and it's default keymap.

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jscotto/scotto36/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Scotto36
+
+![Scotto36](https://imgur.com/a/ZYzRZ9S)
```
Comment (zvecr) [inline keyboards/jscotto/scotto36/README.md:3]:
```suggestion
![Scotto36](https://i.imgur.com/MCGv0ZHh.jpeg
```

**Review by zvecr — CHANGES_REQUESTED** [review]
> A 36 key **handwired** ortholinear ergo keyboard

Needs to be moved into the `keyboards/handwired/` folder

File: keyboards/jscotto/scotto36/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+// Layer Names
+#define _MAC_DEFAULT  0
+#define _MAC_CODE 1
+#define _MAC_NUM 2
+#define _MAC_FUNC 3
+#define _WIN_DEFAULT  4
+#define _WIN_CODE 5
+#define _WIN_NUM 6
+#define _WIN_FUNC 7
```
Comment (fauxpark) [inline keyboards/jscotto/scotto36/keymaps/default/keymap.c:82]: Use an enum here.
```suggestion
enum layer_names {
    _MAC_DEFAULT,
    _MAC_CODE,
    _MAC_NUM,
    _MAC_FUNC,
    _WIN_DEFAULT,
    _WIN_CODE,
    _WIN_NUM,
    _WIN_FUNC
};
```

File: keyboards/jscotto/scotto36/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    TD_ESC_WINDOWS_EMOJI
+};
+
+void td_esc_spotlight_emoji (qk_tap_dance_state_t *state, void *user_data) {
+    if (state->count == 1) {
+        tap_code(KC_ESC);
+    } else if (state->count == 2) {
+        register_code(KC_LGUI);
+        tap_code(KC_SPC);
+        unregister_code(KC_LGUI);
```
Comment (fauxpark) [inline keyboards/jscotto/scotto36/keymaps/default/keymap.c:32]:
```suggestion
        tap_code16(G(KC_SPC));
```

File: keyboards/jscotto/scotto36/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    } else if (state->count == 3) {
+        register_code(KC_LGUI);
+        register_code(KC_LCTL);
+        tap_code(KC_SPC);
+        unregister_code(KC_LGUI);
+        unregister_code(KC_LCTL);
```
Comment (fauxpark) [inline keyboards/jscotto/scotto36/keymaps/default/keymap.c:38]:
```suggestion
        tap_code16(C(G(KC_SPC)));
```

File: keyboards/jscotto/scotto36/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,        KC_CIRC, KC_AMPR, KC_ASTR,                   KC_CAPS, KC_BSPC,
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,           KC_6,    KC_7,    KC_8,                      KC_9,    KC_0,
+        KC_LSFT, KC_NO,   KC_NO,   KC_NO,   MO(3),          KC_NO,   KC_NO,   KC_COMM,                   KC_DOT,  RSFT_T(KC_SLSH),
+                          KC_LCTL, KC_LALT, LGUI_T(KC_SPC), KC_TRNS, KC_TRNS, TD(TD_ESC_SPOTLIGHT_EMOJI)
+    ),
+    [3] = LAYOUT_ortho_3x10_6(
+-        KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,          KC_NO,   KC_NO,   KC_NO,                     KC_NO, TO(4),
+-        KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,          KC_F6,   KC_F7,   KC_F8,                     KC_F9, KC_F10,
+-        KC_F11,  KC_NO, KC_NO,   QK_BOOT, KC_TRNS,        KC_NO,   KC_NO,   KC_NO,                     KC_NO, KC_F12,
+-                        KC_LCTL, KC_LALT, LGUI_T(KC_SPC), KC_TRNS, KC_TRNS, TD(TD_ESC_SPOTLIGHT_EMOJI)
```
Comment (fauxpark) [inline keyboards/jscotto/scotto36/keymaps/default/keymap.c:107]:
```suggestion
         KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,          KC_NO,   KC_NO,   KC_NO,                     KC_NO, TO(4),
         KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,          KC_F6,   KC_F7,   KC_F8,                     KC_F9, KC_F10,
         KC_F11,  KC_NO, KC_NO,   QK_BOOT, KC_TRNS,        KC_NO,   KC_NO,   KC_NO,                     KC_NO, KC_F12,
                         KC_LCTL, KC_LALT, LGUI_T(KC_SPC), KC_TRNS, KC_TRNS, TD(TD_ESC_SPOTLIGHT_EMOJI)
```

File: keyboards/jscotto/scotto36/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    [6] = LAYOUT_ortho_3x10_6(
+        KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR,                 KC_CAPS, KC_BSPC,
+        KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,                    KC_9,    KC_0,
+        KC_LSFT, KC_NO,   KC_NO,   KC_NO,   MO(7),   KC_NO,   KC_NO,   KC_COMM,                 KC_DOT,  RSFT_T(KC_SLSH),
+                          KC_LALT, KC_LCTL, KC_SPC,  KC_TRNS, KC_TRNS, TD(TD_ESC_WINDOWS_EMOJI)
+    ),
+    [7] = LAYOUT_ortho_3x10_6(
+-        KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,                   KC_NO, TO(0),
+-        KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,                   KC_F9, KC_F10,
+-        KC_F11,  KC_NO, KC_NO,   QK_BOOT, KC_TRNS, KC_NO,   KC_NO,   KC_NO,                   KC_NO, KC_F12,
```
Comment (fauxpark) [inline keyboards/jscotto/scotto36/keymaps/default/keymap.c:130]:
```suggestion
         KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,                   KC_NO, TO(0),
         KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,                   KC_F9, KC_F10,
         KC_F11,  KC_NO, KC_NO,   QK_BOOT, KC_TRNS, KC_NO,   KC_NO,   KC_NO,                   KC_NO, KC_F12,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/jscotto/scotto36/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Scotto36
```
[the remark this answers is not in the document]
Comment (joe-scotto) [reply keyboards/jscotto/scotto36/README.md:1]: That's the name of the board.

[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/jscotto/scotto36/README.md:1]: I'm talking about the filename.

[the remark this answers is not in the document]
Comment (joe-scotto) [reply keyboards/jscotto/scotto36/README.md:1]: Oh, the file... okay

File: keyboards/handwired/jscotto/scotto36/keymaps/default/oled.h [hunk 8]
```diff
@@ -0,0 +1,167 @@
+/*
```
Comment (fauxpark) [inline keyboards/handwired/jscotto/scotto36/keymaps/default/oled.h:1]: This code should be in a .c file. The .h file should be separate.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto36/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+         KC_NO,   KC_NO, KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,                   KC_NO, TO(0),
+         KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,                   KC_F9, KC_F10,
+         KC_F11,  KC_NO, KC_NO,   QK_BOOT, KC_TRNS, KC_NO,   KC_NO,   KC_NO,                   KC_NO, KC_F12,
+                         KC_LALT, KC_LCTL, KC_SPC,  KC_TRNS, KC_TRNS, TD(TD_ESC_WINDOWS_EMOJI)
+     )
+};
+
+#if OLED_ENABLE
+    #include "oled.h"
+#endif
```
Comment (fauxpark) [inline keyboards/handwired/jscotto/scotto36/keymaps/default/keymap.c:130]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto36/keymaps/default/keymap.c (hunk 9 above)
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto36/keymaps/default/keymap.c:130]: Doing this completely kills the compilation.

File: keyboards/handwired/jscotto/scotto36/keymaps/default/rules.mk [hunk 10]
```diff
@@ -0,0 +1,4 @@
+OLED_ENABLE = yes
+WPM_ENABLE = yes
+LTO_ENABLE = yes
+TAP_DANCE_ENABLE = yes
```
[the remark this answers is not in the document]
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto36/keymaps/default/rules.mk:4]: Doing this completely kills the compilation.

[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/handwired/jscotto/scotto36/keymaps/default/rules.mk:4]: Then this code will have to live in keymap.c.

[the remark this answers is not in the document]
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto36/keymaps/default/rules.mk:4]: Okay, moved it directly into keymap.c.

**Comment by joe-scotto** [comment]
This should be all set now hopefully.

File: keyboards/jscotto/scotto36/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/jscotto/scotto36/README.md]: This still needs renaming.

[the remark this answers is not in the document]
Comment (joe-scotto) [reply keyboards/jscotto/scotto36/README.md]: Whoops, thought I got that one... should be fixed now.

**Review by fauxpark — APPROVED** [review]

**Comment by joe-scotto** [comment]
@fauxpark What's the next step for this, will @zvecr have to approve as well?

File: keyboards/handwired/jscotto/scotto36/readme.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+![Scotto36](https://i.imgur.com/MCGv0ZHh.jpeg)
+
+A 36 key handwired ortholinear ergo keyboard. Featuring an OLED display with Bongo Cat. Case files available [here](https://github.com/joe-scotto/keyboards.git).
+
+-   Keyboard Maintainer: [Joe Scotto](https://github.com/joe-scotto)
+-   Hardware Supported: ATmega32U4
+-   Hardware Availability: [Amazon](https://amazon.com)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/handwired/jscotto/scotto36/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/jscotto/scotto36/readme.md (hunk 11 above)
Comment (joe-scotto) [reply keyboards/handwired/jscotto/scotto36/readme.md:14]: Fixed.

**Review by drashna — APPROVED** [review]

**Pull request merged**